### PR TITLE
Milestone 2

### PR DESCRIPTION
**Pull request by asiimwedismas**
In this milestone, we created a model with associations and all requested features for friendships.

### CODE REVIEW

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hello!:wave: Good start on this milestone:+1: Here are your required improvements:**

:point_right: A user should be able to invite other users to friendship. This option should be available on the all users page and on each user's show page.

:point_right: A user who receives a friend request should have the option of accepting or rejecting the invitation.

:point_right: Timeline should have posts of the logged-in user and their friends, not all posts.

:point_right: Add unit tests for your model associations and validations and integration tests for your features.

:point_right: Extract view logic in posts index into helper methods.


_**Kindly make these changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested:exclamation:

**Comment by asiimwedismas** [comment]
> **Hello!👋 Good start on this milestone👍 Here are your required improvements:**
> 
> 👉 A user should be able to invite other users to friendship. This option should be available on the all users page and on each user's show page.
> 
> 👉 A user who receives a friend request should have the option of accepting or rejecting the invitation.
> 
> 👉 Timeline should have posts of the logged-in user and their friends, not all posts.
> 
> 👉 Add unit tests for your model associations and validations and integration tests for your features.
> 
> 👉 Extract view logic in posts index into helper methods.
> 
> _**Kindly make these changes and submit another code review request.**_
> 
> **Any questions? You can reach me on Slack (Moses Ogwo)**
> 
> **Cheers🌠**
> 
> ### Status: Changes Requested❗

All issues raised have been fixed...👍👍👍👍👍

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
Hello Team :wave: 
Status: Changes Required :recycle: 

Great job on doing the changes :+1: 
Your website's functionality is amazing! Everything works fine...
Let's cover some details together :arrow_down: 

As the previous TSE said:

- [x] [REQUIRED] **Extract the logic in posts index view** into helper methods.
:bulb: In your app/views/posts/index file => you have `if block`. All the logic should be separated from the views. So let's move it into a helper method that you will create from scratch. And then you will only call that method in the view :blush: 

- [x] [REQUIRED]  Add unit tests for your model associations and validations and integration tests for your features.
:bulb: **Please add more tests**. :pray:  You have very few test cases for now, but I am sure you can come up with a lot of cases in which you are trying different aspects of your website. 

## That's it! After doing the changes, you can request for another review :blush: 

With all my respect :pray:

**Comment by asiimwedismas** [comment]
> Hello Team 👋
> Status: Changes Required ♻️
> 
> Great job on doing the changes 👍
> Your website's functionality is amazing! Everything works fine...
> Let's cover some details together ⬇️
> 
> As the previous TSE said:
> 
> * [x]  [REQUIRED] **Extract the logic in posts index view** into helper methods.
>   💡 In your app/views/posts/index file => you have `if block`. All the logic should be separated from the views. So let's move it into a helper method that you will create from scratch. And then you will only call that method in the view 😊
> * [x]  [REQUIRED]  Add unit tests for your model associations and validations and integration tests for your features.
>   💡 **Please add more tests**. 🙏  You have very few test cases for now, but I am sure you can come up with a lot of cases in which you are trying different aspects of your website.
> 
> ## That's it! After doing the changes, you can request for another review 😊
> With all my respect 🙏

All sorted out

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -4,7 +4,18 @@
...
+          
+          <% if current_user != User.find(user.id)%>
+            <% if show_add_friend(user) %>
+              <%= link_to 'Invite to friendship', {:controller => 'friendships', :action => 'create', :id => user.id}, method: :post, class: 'profile-link' %>
+            <% end %>
+
+            <% if show_already_friend(user) %>
+              <span> | you are friends</span>
+            <% end %>
+          <% end %>
```
Comment (kobiyoyo) [inline app/views/users/index.html.erb]: Extract the logic in Users view into helper methods.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+
+  <% if current_user != User.find(@user.id)%>
+
+    <% if show_add_friend(@user) %>
+      <%= link_to 'Invite to friendship', {:controller => 'friendships', :action => 'create', :id => @user.id}, method: :post, class: 'profile-link' %>
+    <% end %>
+
+    <% if show_already_friend(@user) %>
+      <span> | you are friends</span>
+    <% end %>
```
Comment (kobiyoyo) [inline app/views/users/show.html.erb]: Extract the logic in Users view into helper methods.

File: app/views/friendships/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+              <%= link_to 'Invite to friendship', {:controller => 'friendships', :action => 'create', :id => user.id}, method: :post, class: 'profile-link' %>
+            <% end %>
+
+            <% if show_already_friend(user) %>
+              <span> | you are friends</span>
+            <% end %>
+          <% end %>
+        </span>
+      </li> 
+    <% end %>
```
Comment (kobiyoyo) [inline app/views/friendships/index.html.erb]: Extract the logic in friendships view into helper methods.

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [x] Check comments below
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: app/views/users/index.html.erb (hunk 1 above)
Comment (CalyCherkaoui) [reply app/views/users/index.html.erb]: Issue solved

File: app/views/users/show.html.erb (hunk 2 above)
Comment (CalyCherkaoui) [reply app/views/users/show.html.erb]: Issue solved

File: app/views/friendships/index.html.erb (hunk 3 above)
Comment (CalyCherkaoui) [reply app/views/friendships/index.html.erb]: Issue solved

**Review by kubilaycaglayan — APPROVED** [review]
Hello Team :wave: 
Status: Milestone Approved :green_circle: 

Amazing job :100: :100: :100: 
Isn't it great? Now you know more than yesterday! :star: 

Thanks for your efforts and time 🙏
With all my respect...

**Pull request merged**